src/main/scala/com/campudus/tableaux/database/model/TableauxModel.scala

```diff
@@ -150,12 +203,33 @@ class TableauxModel(override protected[this] val connection: DatabaseConnection)
     }
 
     for {
-      rawRow <- rowModel.retrieve(column.table.id, rowId, columns)
-      rowSeq <- mapRawRows(column.table, columns, Seq(rawRow))
+      valueCache <- CACHING match {
+        case true => CacheClient(this.connection.vertx).retrieveCellValue(column.table.id, column.id, rowId)
+        case false => Future.successful(None)
+      }
+
+      value <- valueCache match {
+        case Some(obj) =>
+          // Cache hit
+          Future.successful(obj)
+        case None => {
+          // Cache miss
+          for {
+            rawRow <- rowModel.retrieve(column.table.id, rowId, columns)
+
+            rowSeq <- mapRawRows(column.table, columns, Seq(rawRow))
+
+            // Because we only want a cell's value other
+            // potential rows and columns can be ignored.
+            value = rowSeq.head.values.head
 
-      // Because we only want a cell's value other
-      // potential rows and columns can be ignored.
-      value = rowSeq.head.values.head
+            _ <- CACHING match {
```

Comment:
You could do `_ = if (CACHING) { sideEffectingFunction() }` if you use the reference in "Kartoffelpürre".